File: Python/0032. 整数拆分 Ver.2.py
# If you need to import additional packages or classes, please import here.
def func():
    # please define the python3 input here. 
    # For example: a,b = map(int, input().strip().split())
    # please finish the function body here.
    # please define the python3 output here. For example: print().
    while True:
        try:
            n = int(input())        
            def dfs(num,index,s):                
                if num == 0:
                    print(str(n)+'='+s[1:])                   
                for i in range(index,num+1):
                    dfs(num-i,i,s+"+"+str(i))            
            dfs(n,1,"")
       
        except EOFError:
            break
    
if __name__ == "__main__":
    func()

File: C++/0032. 整数拆分 Ver.2.cpp
// we have defined the necessary header files here for this problem.
// If additional header files are needed in your program, please import here.

int n;

void dfs(int num,int index,string s){
    if(num==0){
        cout<<n<<'='<<s.substr(1)<<endl;
        return;
    }
    for(int i=index;i<=num;++i){
        dfs(num-i,i,s+'+'+to_string(i));
    }            
}

int main()
{  
  // please define the C++ input here. For example: int a,b; cin>>a>>b;;  
  // please finish the function body here.  
  // please define the C++ output here. For example:cout<<____<<endl; 
    while(cin>>n){
        dfs(n,1,"");
    }
    return 0;
}
